libsolidity/ast/Types.cpp

```diff
@@ -3006,7 +3006,17 @@ string FunctionType::canonicalName() const
 
 string FunctionType::toString(bool _short) const
 {
-	string name = "function ";
+	string name;
+	switch (m_kind)
+	{
+	case Kind::Declaration:
+	case Kind::Internal:
+	case Kind::External:
+		name = "function ";
+		break;
+	default:
+		name = "builtin function ";
```

Comment:
This is a breaking change for the json-AST :(

Reply:
Or maybe it isn't...
On the other hand: Maybe we should also output the name of the builtin function instead of its signature?

Reply:
I'm fine with that, this function in general is terrible and would be ready for a rewrite.

Reply:
Having a second look at `FunctionType::richIdentifier` it is **a lot of them**. Not sure it is worth it?

Reply:
can't you extract the switch from `richIdentifier` and re-use it?

Reply:
@chriseth what do you suggest to be displayed for builtin functions?

Reply:
Either `builtin function keccak256` or `keccak256` or `function keccak256`. What do you do about `abi.encode`?

Reply:
In this new version it would display `builtin abiencode`.